kentik_api_library/kentik_api/public/custom_application.py

```diff
@@ -42,7 +42,7 @@ def __init__(
     @property
     def id(self) -> ID:
         if self._id is None:
-            raise IncompleteObjectError("Custom Application", "_id is required")
+            raise IncompleteObjectError("", self.__class__.__name__, "_id is required")
```

Comment:
Probably another error should be raised here, like InvalidStateError, see comment below

Reply:
I think `IncompleteObjectErrors` fits the situation reasonably well.